d2l-input-search.html

```diff
@@ -217,6 +195,29 @@
 				if (this.value !== undefined && this.value !== null) {
 					this._setLastSearchValue(this.value);
 				}
+				this._handleFocus = this._handleFocus.bind(this);
+				this._handleBlur = this._handleBlur.bind(this);
+				this._handleMouseEnter = this._handleMouseEnter.bind(this);
+				this._handleMouseLeave = this._handleMouseLeave.bind(this);
+			},
+
```

Comment:
@dbatiste Interestingly, I had to put the explicit event handlers back. Without them, Polymer 1 doesn't get the correct source of the events.

Reply:
I've run into other issues with the `listeners` block before, and in Polymer 2 they don't even recommend using it (https://meowni.ca/posts/polymer-2-cheatsheet/#listeners).

Reply:
I was thinking about this a bit last night.  I think the reason `useCapture = true` was/is being used here is because these events don't bubble from the target.  To accomplish event delegation for them, `useCapture` must be used.  The alternative is to use different events.

The other thing that can play into this is how focus is tracked when shadow-DOM is enabled.  I started to write a wack of stuff but then just figured it would be better to pass along Sam Thorogood's [Focus inside shadow-DOM article](https://medium.com/dev-channel/focus-inside-shadow-dom-78e8a575b73). 

I think I would need to take a closer look to be absolutely certain, but I think these two things (the capture/bubbling & the effect of shadow-DOM on focus) may explain why things were not working when the listeners are wired up at the host level without capturing, and the differences between P1 + P2.